hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +48,284 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+    // check for collision warnings
+    checkCollisionWarnings(timestep);
+
     // set random grid shift
     m_cl->drawGridShift(timestep);
 
     // update cell list
     m_cl->compute(timestep);
 
+    // create auxillary array for rigid bodies
+    if (m_rigid_bodies)
```

Comment:
These should be like this:
```suggestion
    const bool rigid_body_collision = m_embed_group && m_rigid_bodies;
    if (rigid_body_collision)
```
You only need to do these steps if you have an embedded group AND rigid bodies.